pkg/agent/params.go

```diff
@@ -102,75 +105,94 @@ func assignKubernetesParametersFromAgentProfile(profile *datamodel.AgentPoolProf
 	}
 }
 
-func assignKubernetesParameters(properties *datamodel.Properties, parametersMap paramsMap,
+func assignKubernetesParametersfromKubernetesConfig(properties *datamodel.Properties, parametersMap paramsMap,
 	cloudSpecConfig *datamodel.AzureEnvironmentSpecConfig,
 	k8sComponents *datamodel.K8sComponents,
-	generatorCode string,
-	config *datamodel.NodeBootstrappingConfiguration) {
+	config *datamodel.NodeBootstrappingConfiguration) paramsMap {
```

Comment:
I think I asked this before, but do we even need to return anything out of this func since it's just mutating in-place?

though not sure if the linter would like empty returns or not in short-circuit cases...